src/routes/devices/tags.test.ts

```diff
@@ -0,0 +1,73 @@
+import { getDistinctTags } from './tags'
+import { logger as log } from '../../utils/logger'
+
+let res: Express.Response
+let statusSpy: jest.SpyInstance
+let jsonSpy: jest.SpyInstance
+
+beforeEach(() => {
+  res = {
+    status: () => {
+      return res
+    },
+    json: () => {
+      return res
+    },
+    end: () => {
+      return res
+    }
+  }
+  statusSpy = jest.spyOn(res as any, 'status')
+  jsonSpy = jest.spyOn(res as any, 'json')
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+  jest.resetModules()
+})
+
+describe('tags', () => {
+  it('should set status to 200 when getting results from getDistinctTags', async () => {
+    const expectedResult = {}
+    const req = {
```

Comment:
just a suggestion, could put the defintion of req in the beforeEach to reduce code duplication and then just set req.db.devices.getDistinctTags when it needs to change (ie. only in error calls)

Reply:
refactored to remove duplication in next commit